### DISCUSSION TITLE
Javascript: tanta coisa, por onde começar?

### DISCUSSION THREAD

**Discussion by kvnol** [frontendbr/forum#270] [Perguntas e Respostas] [answered]
Estou na web desde 2009 me apaixonando cada dia mais por códigos, mas por conta de outros projetos (escola, na época, depois trabalho e música — paixão) parei de mexer com web, até que arrumei um trampo como Front-end Jr e estou quebrando a cabeça com o Javascript.

A dúvida é: como eu posso me localizar para estudar no meio de tantas tecnologias, me sinto atrasado em js, mas estou relativamente avançado em HTML e CSS. Sou um verdadeiro "recortador de layout".

**Comment by woliveiras** [comment]
@kvnol, você já está de boas em relação a lógica de programação?

**Comment by expalmer** [comment, marked as answer]
Não se assustem nem se preocupe agora com frameworks e coisas do tipo (React, Angular, NodeJs, Gulp, Grunt, WebPack)...seja paciente.
Aprenda os conceitos, os design patterns, conheça o que é o Javascript. Depois disso qualquer framework vai ser tranquilo de entender e muito mais fácil de usar.
Em geral muitos tentam usar as últimas ferramentas do mercado, sem entender como elas funcionam, e isso porque não entendem Javascript, apenas sabem como "usar" uma determinada ferramenta.

Sugiro esse caminho:

[JsTheRightWay](http://jstherightway.org/pt-br/)

Se você for dedicado, logo estará bem seguro em JS.

**Comment by kvnol** [comment]
@woliveiras sim, só preciso dar uma revisada em algoritmos mas já tenho uma boa noção. Iria até perguntar se tem algum site com exercícios para eu testar minha lógica de programação.

**Comment by ludwiklejzer** [comment]
Estou lendo o [You Don't Know JS](https://github.com/cezaraugusto/You-Dont-Know-JS), é incrível! Como sou iniciante está me ajudando bastante a compreender o JS. Dá uma olhada, acho que lhe ajudará bastante.

**Comment by kvnol** [comment]
Porra, @expalmer. Ducaralho! E sim, não tô me preocupando muito com os frameworks e quero sofrer com o js puro, o link já está salvo e vou começar a estudar urgentemente.

E, @LajackBR, salvei o link também! Muito obrigado, galera.

**Comment by herberthenrique** [comment]
Bem, eu começaria por lógica de programação.
Estude um pouco da sintaxe do JS, nesses links que mandaram e depois aplique aqui:
http://www.devfuria.com.br/logica-de-programacao/

Inclusive, eles fazem uma intro ao [projeto euler](https://projecteuler.net/), um dos testes e desafios de logica mais dificeis e divertidos que eu já fiz

**Comment by camilatigre** [comment]
@kvnol tem exércicios de JS [nesse site](http://www.codewars.com/). Eu usava ele e gostava ((:

**Comment by leandroruel** [comment]
leia o blog do meu mestre: http://desenvolvimentoparaweb.com/javascript/javascript-iife-conteiner-de-codigos/

**Comment by lucasengel** [comment]
How to Learn JS Properly: http://javascriptissexy.com/how-to-learn-javascript-properly/

**Comment by felipemrodrigues** [comment]
Lógica de programação, algoritmos, estrutura de dados.

Falando mais especificamente de JS, gosto do [Eloquent Javascript](https://www.amazon.com.br/Eloquent-JavaScript-Modern-Introduction-Programming-ebook/dp/B00QL616UU/ref=sr_1_1?ie=UTF8&qid=1470231703&sr=8-1&keywords=eloquent+javascript). Parece que existe uma versão gratuita na web, inclusive em português.

**Comment by gabrieldarezzo** [comment]
Apesar de você já ter uma noção, tem alguns exercícios no meu repo e alguns links. (Inclusive peguei alguns e já dei um update lá hehe)

https://github.com/gabrieldarezzo/helpjs-ravi/blob/master/README.md#artigos-lekais-de-javascripthtmlcss

**Comment by rubenmarcus** [comment]
Vale a pena começar pelo ES6 já?

**Comment by fdaciuk** [comment]
Lógica de programação não faz muito sentido. O que se ensina em "lógica de programação" é a forma imperativa de fazer as coisas. JavaScript é uma linguagem multiparadigma. Você consegue programar de forma imperativa, mas você também programar orientado a objetos e de forma funcional. Tudo no mesmo código, usando o que tem de melhor em cada paradigma.

Estude e entenda como o JavaScript funciona. Quando você pegar outra linguagem para aprender, vai descobrir coisas semelhantes ao JavaScript, e até coisas que você pode aplicar nos seus programas em JavaScript.

Então, minha dica é: conheça o **core** da linguagem, saiba como ela funciona por debaixo dos panos, que você irá conseguir utilizar qualquer framework que aparecer =)

---

Sobre começar com ES6: a maior parte das features do ES6 são _sintax sugar_ pra coisas que você consegue fazer com ES5. Então, estude JS a partir do ES5, e depois estude ES6, para entender o porquê das melhorias que ele propõe.

**Comment by R16-1000** [comment]
boa noite
Estou començando começando agora no curso de JavaScript e gostaria de conselhos e também ajuda com dificuldades ao decorrer do curso.

**Reply by ROBERTOCONRADO** [reply]
Tente aprender javascript puro primeiro antes de framework

**Comment by josuebsilva** [comment]
Eu criei um canal que pode ajudar. Ensino desenvolvimento web a programação. Confira o meu canal e blog
Site: [Jedibit.com.br](https://jedibit.com.br) e nesse vídeo eu dou uma breve introdução ao [Javascript](https://youtu.be/7gI7vjs37tk)
